implementations/rust/ockam/ockam_command/src/trust_context/delete.rs

```diff
@@ -0,0 +1,65 @@
+use anyhow::anyhow;
+use clap::Args;
+use colorful::Colorful;
+
+use ockam_api::cli_state::traits::StateDirTrait;
+use ockam_api::cli_state::CliStateError;
+
+use crate::terminal::ConfirmResult;
+use crate::{docs, fmt_ok, fmt_warn, CommandGlobalOpts};
+
+const LONG_ABOUT: &str = include_str!("./static/delete/long_about.txt");
+const AFTER_LONG_HELP: &str = include_str!("./static/delete/after_long_help.txt");
+
+/// Delete a trust context
+#[derive(Clone, Debug, Args)]
+#[command(
+    arg_required_else_help = false,
+    long_about = docs::about(LONG_ABOUT),
+    after_long_help = docs::after_help(AFTER_LONG_HELP)
+)]
+pub struct DeleteCommand {
+    /// Name of the trust context
+    pub name: String,
+}
+
+impl DeleteCommand {
+    pub fn run(self, opts: CommandGlobalOpts) {
+        if let Err(e) = run_impl(opts, self) {
+            eprintln!("{e:?}");
+            std::process::exit(e.code());
+        }
+    }
+}
+
+fn run_impl(opts: CommandGlobalOpts, cmd: DeleteCommand) -> crate::Result<()> {
+    let DeleteCommand { name } = cmd;
+    let state = opts.state.trust_contexts;
+    match state.get(&name) {
+        // If it exists, proceed
+        Ok(_) => {
+            if let ConfirmResult::No = opts.terminal.confirm(&fmt_warn!(
```

Comment:
Should we add a `--yes` option every time we require a user confirmation?

Reply:
Yeah... I think we should have the same as we currently have in the `ockam reset` command:
- TTY: ask for confirmation if `--yes` is not passed
- Non TTY: fail if `--yes` is not passed

Reply:
Yeah... I think we should have the same as we currently have in the `ockam reset` command:
- TTY: ask for confirmation if `--yes` is not passed
- Non TTY: fail if `--yes` is not passed

I'll tackle this in a separate PR

Reply:
I've created an issue for that https://github.com/build-trust/ockam/issues/5041